sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/Trigger.java

```diff
@@ -263,13 +264,16 @@
     public abstract MergingTriggerInfo trigger();
   }
 
-  @Nullable
-  protected final List<Trigger> subTriggers;
+  public final List<Trigger> subTriggers;
```

Comment:
The list itself is public? That seems broken.

I would suggest getting rid of the support for subtriggers in the base class, and just have each trigger that receives subtriggers special case it. The translation/execution at TriggerStateMachine is going to be pattern-matching on the kind of trigger anyways.